examples/RodContactCase/RodSelfContact/PlectonemesCase/plectoneme_case.py

```diff
@@ -0,0 +1,218 @@
+
+
+plectonemes_sim.append(sherable_rod)
+
+# boundary condition
+from elastica._rotations import _get_rotation_matrix
+
+
+class SelonoidsBC(FreeRod):
+    """
+
+    """
+
+    def __init__(
+        self,
+        position_start,
+        position_end,
+        director_start,
+        director_end,
+        twisting_time,
+        time_twis_start,
+        number_of_rotations,
+    ):
+        FreeRod.__init__(self)
+        self.twisting_time = twisting_time
+        self.time_twis_start = time_twis_start
+
+        theta = 2.0 * number_of_rotations * np.pi
+
+        angel_vel_scalar = theta / self.twisting_time
+
+        direction = -(position_end - position_start) / np.linalg.norm(
+            position_end - position_start
+        )
+
+        axis_of_rotation_in_material_frame = director_end @ direction
+        axis_of_rotation_in_material_frame /= np.linalg.norm(
+            axis_of_rotation_in_material_frame
+        )
+
+        self.final_end_directors = (
+            _get_rotation_matrix(
+                -theta, axis_of_rotation_in_material_frame.reshape(3, 1)
+            ).reshape(3, 3)
+            @ director_end
+        )  # rotation_matrix wants vectors 3,1
+
+        self.ang_vel = angel_vel_scalar * axis_of_rotation_in_material_frame
+
+        self.position_start = position_start
+        self.director_start = director_start
+
+    def constrain_values(self, rod, time):
+        if time > self.twisting_time + self.time_twis_start:
+            rod.position_collection[..., 0] = self.position_start
+            rod.position_collection[0, -1] = 0.0
+            rod.position_collection[2, -1] = 0.0
+
+            rod.director_collection[..., 0] = self.director_start
+            rod.director_collection[..., -1] = self.final_end_directors
+
+    def constrain_rates(self, rod, time):
+        if time > self.twisting_time + self.time_twis_start:
+            rod.velocity_collection[..., 0] = 0.0
+            rod.omega_collection[..., 0] = 0.0
+
+            rod.velocity_collection[..., -1] = 0.0
+            rod.omega_collection[..., -1] = 0.0
+
+        elif time < self.time_twis_start:
+            rod.velocity_collection[..., 0] = 0.0
+            rod.omega_collection[..., 0] = 0.0
+
+        else:
+            rod.velocity_collection[..., 0] = 0.0
+            rod.omega_collection[..., 0] = 0.0
+
+            rod.velocity_collection[0, -1] = 0.0
+            rod.velocity_collection[2, -1] = 0.0
+            rod.omega_collection[..., -1] = -self.ang_vel
+
+            rod.velocity_collection[2, int(rod.n_elems / 2)] -= 1e-4
+
+
+plectonemes_sim.constrain(sherable_rod).using(
+    SelonoidsBC,
+    constrained_position_idx=(0, -1),
+    constrained_director_idx=(0, -1),
+    time_twis_start=time_start_twist,
+    twisting_time=time_twist,
+    number_of_rotations=number_of_rotations,
+)
+
+# Add self contact to prevent penetration
+plectonemes_sim.connect(sherable_rod, sherable_rod).using(SelfContact, k=1e4, nu=10)
```

Comment:
Ah I see. This interface is not ideal and seems like an abuse of our DSL. "Connect" a rod to itself seems bizarre to me.

Ideally, all contacts are lumped in Environments. And we use a flag like
`sim.set_collision(on)`
to enable/disable it. But again, this can be considered in the next release.

Reply:
Okay that is a good idea. I will create a new issue for that.